packages/E2ETest/windows/ReactUWPTestApp/ReactUWPTestApp.csproj

```diff
@@ -165,24 +165,13 @@
   <ItemGroup>
     <Folder Include="Themes\" />
   </ItemGroup>
-  <ItemGroup>
-    <ProjectReference Include="$(ReactNativeWindowsDir)\Microsoft.ReactNative\Microsoft.ReactNative.vcxproj">
-      <Project>{f7d32bd0-2749-483e-9a0d-1635ef7e3136}</Project>
-      <Name>Microsoft.ReactNative</Name>
-    </ProjectReference>
-    <ProjectReference Include="$(ReactNativeWindowsDir)\Microsoft.ReactNative.Managed\Microsoft.ReactNative.Managed.csproj">
-      <Project>{F2824844-CE15-4242-9420-308923CD76C3}</Project>
-      <Name>Microsoft.ReactNative.Managed</Name>
-    </ProjectReference>
-  </ItemGroup>
   <PropertyGroup Condition=" '$(VisualStudioVersion)' == '' or '$(VisualStudioVersion)' &lt; '16.0' ">
     <VisualStudioVersion>16.0</VisualStudioVersion>
   </PropertyGroup>
   <Import Project="$(MSBuildExtensionsPath)\Microsoft\WindowsXaml\v$(VisualStudioVersion)\Microsoft.Windows.UI.Xaml.CSharp.targets" />
-  <Import Project="$(ReactNativeWindowsDir)\PropertySheets\Bundle.props" />
+  <Import Project="$(ReactNativeWindowsDir)\PropertySheets\external\Microsoft.ReactNative.Uwp.CSharpApp.props"/>
```

Comment:
After TONS of debugging, reading verbose msbuild logs. It turns out that <ProjectDir> is never defined by any managed project. This is something controlled by the cpp targets. But in the managed world, Microsoft.Windows.UI.Xaml.CSharp ensures its defined. 
Therefore the props and targets always must be added AFTER that target. I don't think that when we place these props and targets in a nuget package, we will get inserted in the right macro-expansion order.

Should we:
1. Leave as is
1. Throw error from CSharpApp.props if ProjectDir is not deinfed
1. Update CSharpApp.props to define ProjectDir if it is not yet defined.

Reply:
To me the option (3) sounds as the best one since some targets like Microsoft.Windows.UI.Xaml.CSharp do the same.

Reply:
sounds good. Also crated a common props since we were getting too much duplication.

Reply:
@dannyvv `$(MSBuildProjectDirectory)` is the standard way to do this